client/settings/settings-tab-panel/test/index.test.js

```diff
@@ -0,0 +1,32 @@
+/**
```

Comment:
What do you think about:
- Also adding a test that ensures that when the `panel` query string is `settings`, the rendered content is the setting's panel
- Ensuring all these tests are wrapped by a `describe('UpeSettingsTabPanel`,...)`

Reply:
I mocked the `@woocommerce/navigation` module to take care of this:

https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1720/files#diff-b40ba71418bd6a8d99f1f4b126501b336260276abe8a5b71c67cdf9e1997b73fR7

but I don't like the way I had to do it to get it to work:

```
jest.mock( '@woocommerce/navigation', () => ( {
	getQuery: jest.fn().mockReturnValue( { panel: 'settings' } ),
} ) );
```

This assumes that query string for _all_ tests. In this case it's not a big deal but I'm pulling my hair out trying to figure out why this isn't working and need to figure out what I'm doing wrong. -_-

What I was _trying_ to do was mock the module like this:
```
import { getQuery } from '@woocommerce/navigation';
jest.mock( '@woocommerce/navigation', () => ( {
	getQuery: jest.fn(),
} ) );
```

Then, inside of a specific test, I could mock the return value, like so:

`getQuery.mockReturnValue( { panel: 'settings' } );`

But `getQuery` always ends up as undefined. 

I also tried setting the mock to initially have a return value of an empty object, like so:

```
import { getQuery } from '@woocommerce/navigation';
jest.mock( '@woocommerce/navigation', () => ( {
	getQuery: jest.fn( () => ( {} ) ),
} ) );
```

and then in a test mocking the return value to `{ panel: 'settings' }` but that didn't work either.
And lastly, I tried directly mocking the return value in the module mock and overriding it per-test:

```
import { getQuery } from '@woocommerce/navigation';
jest.mock( '@woocommerce/navigation', () => ( {
	getQuery: jest.fn().mockReturnValue( {} ),
} ) );
```

`getQuery.mockReturnValue( { panel: 'settings' } );`

I still keep getting `undefined` for the value of `getQuery()`.

What is also tripping me up here is that I'm not calling `getQuery()` directly in my tests but rather it's being called within a component so I'm not sure if there is something I'm not understanding about how that should be done.

🤷

Reply:
What do you think about this: https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1734/files

Unfortunately what you're trying to do is to click on the tabs but not changing the expected value of the URL via `getQuery()`. In the original code, `getQuery()` was being executed at the root level, so no matter how many times you were mocking the value, it would have always obtained the "initial" value of when the file was included.

Reply:
Unfortunately the branch's tests aren't working either. I fixed the object destructuring fail and I think we should be happy enough with this PR as is, I can't seem to find a way to get this perfect and the effort isn't worth the miniscule improvement. I'd rather merge so we can move on and not block other issues. 👍

Reply:
Weird, they seem to be working for me when running them:
![Screen Shot 2021-08-25 at 8 36 18 AM](https://user-images.githubusercontent.com/273592/130801223-bdab4b0f-0e7d-4fbb-a1c6-79afcdca4f3d.png)


I imagine maybe you're referring to the GH Actions failure? I suspect that's happening because of the error at the time I forked the branch, they seem to match: https://github.com/woocommerce/woocommerce-gateway-stripe/runs/3411147914

I rebased the changes on the branch on top of yours again and the CI statuses match again 👍